ocean_lib/services/service.py

```diff
@@ -272,3 +299,73 @@ def update_compute_values(
         ] = trusted_algo_publishers
         self.compute_values["allowNetworkAccess"] = allow_network_access
         self.compute_values["allowRawAlgorithm"] = allow_raw_algorithm
+
+
+class ConsumerParameters:
+    def __init__(
+        self,
+        name: str,
+        type: Optional[str] = None,
+        label: Optional[str] = None,
+        required: Optional[bool] = False,
+        default: Optional[str] = None,
+        description: Optional[str] = None,
```

Comment:
I think for required the default is False, in case it is missing. But otherwise when creating the customerParameters, I would opt for the most flexibility possible. I do think only the name should be required, but I agree, let's wait for the discussion in the docs repo.

Reply:
The issue and related PR have been merged: https://github.com/oceanprotocol/docs/issues/1005.

All the consumerParameter elements are required except `options`